pkg/strace/syscall_linux_arm64.go

```diff
@@ -0,0 +1,387 @@
+	unix.SYS_PREADV:            makeSyscallInfo("preadv", Hex, ReadIOVec, Hex, Hex),
+	unix.SYS_PWRITEV:           makeSyscallInfo("pwritev", Hex, WriteIOVec, Hex, Hex),
+	unix.SYS_RT_TGSIGQUEUEINFO: makeSyscallInfo("rt_tgsigqueueinfo", Hex, Hex, Hex, Hex),
+	unix.SYS_PERF_EVENT_OPEN:   makeSyscallInfo("perf_event_open", Hex, Hex, Hex, Hex, Hex),
+	unix.SYS_RECVMMSG:          makeSyscallInfo("recvmmsg", Hex, Hex, Hex, Hex, Hex),
+	unix.SYS_FANOTIFY_INIT:     makeSyscallInfo("fanotify_init", Hex, Hex),
+	unix.SYS_FANOTIFY_MARK:     makeSyscallInfo("fanotify_mark", Hex, Hex, Hex, Hex, Hex),
+	unix.SYS_PRLIMIT64:         makeSyscallInfo("prlimit64", Hex, Hex, Hex, Hex),
+	unix.SYS_NAME_TO_HANDLE_AT: makeSyscallInfo("name_to_handle_at", Hex, Hex, Hex, Hex, Hex),
+	unix.SYS_OPEN_BY_HANDLE_AT: makeSyscallInfo("open_by_handle_at", Hex, Hex, Hex),
+	unix.SYS_CLOCK_ADJTIME:     makeSyscallInfo("clock_adjtime", Hex, Hex),
+	unix.SYS_SYNCFS:            makeSyscallInfo("syncfs", Hex),
+	unix.SYS_SENDMMSG:          makeSyscallInfo("sendmmsg", Hex, Hex, Hex, Hex),
+	unix.SYS_SETNS:             makeSyscallInfo("setns", Hex, Hex),
+	unix.SYS_GETCPU:            makeSyscallInfo("getcpu", Hex, Hex, Hex),
+	unix.SYS_PROCESS_VM_READV:  makeSyscallInfo("process_vm_readv", Hex, ReadIOVec, Hex, IOVec, Hex, Hex),
+	unix.SYS_PROCESS_VM_WRITEV: makeSyscallInfo("process_vm_writev", Hex, IOVec, Hex, WriteIOVec, Hex, Hex),
+	unix.SYS_KCMP:              makeSyscallInfo("kcmp", Hex, Hex, Hex, Hex, Hex),
+	unix.SYS_FINIT_MODULE:      makeSyscallInfo("finit_module", Hex, Hex, Hex),
+	unix.SYS_SCHED_SETATTR:     makeSyscallInfo("sched_setattr", Hex, Hex, Hex),
+	unix.SYS_SCHED_GETATTR:     makeSyscallInfo("sched_getattr", Hex, Hex, Hex),
+	unix.SYS_RENAMEAT2:         makeSyscallInfo("renameat2", Hex, Path, Hex, Path, Hex),
+	unix.SYS_SECCOMP:           makeSyscallInfo("seccomp", Hex, Hex, Hex),
+}
+
+// FillArgs pulls the correct registers to populate system call arguments
+// and the system call number into a TraceRecord. Note that the system
+// call number is not technically an argument. This is good, in a sense,
+// since it makes the function arguments end up in "the right place"
+// from the point of view of the caller. The performance improvement is
+// negligible, as you can see by a look at the GNU runtime.
+func (s *SyscallEvent) FillArgs() {
+	s.Args = SyscallArguments{
+		{uintptr(s.Regs.Regs[1])},
+		{uintptr(s.Regs.Regs[2])},
+		{uintptr(s.Regs.Regs[3])},
+		{uintptr(s.Regs.Regs[4])},
+		{uintptr(s.Regs.Regs[5])},
+		{uintptr(s.Regs.Regs[6])}}
+	s.Sysno = int(uint32(s.Regs.Regs[0]))
+}
+
+// FillRet fills the TraceRecord with the result values from the registers.
+func (s *SyscallEvent) FillRet() {
+	s.Ret = [2]SyscallArgument{{uintptr(s.Regs.Regs[0])}}
+	if errno := int(s.Regs.Regs[0]); errno < 0 {
+		s.Errno = unix.Errno(-errno)
+	}
+}
+
+// Signal table (taken from Go runtime, but adding all-caps signal names)
+var signals = [...]string{
```

Comment:
Surely this is common across all Linux?

Reply:
Can reason this cannot just be `[]string`?

Reply:
perhaps ironically, when I first wrote this code, the internal Google reviewer demanded [...]

It's an arrray of "just the right size", at that point, not a slice.

I moved this to syscall_linux.go. I'm actually shocked that is it the same across all Linux.

Reply:
Y'all are misreading each other. Ryan said "surely this is" - he was asking you.

It is not, actually. See "signal numbering" in https://man7.org/linux/man-pages/man7/signal.7.html

Reply:
Although MIPS might be the only one that Go actually supports in that table...